packages/@aws-cdk/aws-apigateway/lib/stage.ts

```diff
@@ -246,7 +287,7 @@ export class Stage extends Resource {
       if (options.cachingEnabled) {
         if (self.enableCacheCluster === undefined) {
           self.enableCacheCluster = true;
-        } else if (self.enableCacheCluster === false) {
+        } else if (!self.enableCacheCluster) {
```

Comment:
Any reason we're changing this? The old statement looks correct.

Reply:
This old statement is correct, but changed because it is redundant.
Is it out of the scope of the pull request and should it be restored?

Reply:
`!x` will evaluate to true, both when `x` is set to `false` and when set to `undefined`.

In this particular case, there are different behaviours being configured when `enableCacheCluster` is set to `undefined` and when set to `false`. 

The previous statement made this very clear by explicitly checking against `=== undefined` and `=== false`. Let's keep this explicitness in place especially when the behaviour should be different between `undefined` and `false`.

Reply:
I understand, I'll take it back.